tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -55,112 +55,131 @@ steps:
       echo Tag: $tag
       cp .npmrc ~/.npmrc
       maximumRetryIfNetworkError=3
+
+      # Note: the package names here are "processed" the way 'npm pack' does (if the package is scoped, remove
+      # the leading @ and replace / with -).
+      # When getting them from a txt file that's the way we wrote them to it.
+      # When getting them from the output of 'ls' that's also how they are because the files are the 'npm pack'ed
+      # packages but we need to strip the version number and the extension for consistency, so later we can assume
+      # that these are just the package names, as we need them like that for some things.
       if [[ -f $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt ]]; then
-        packages=$(cat $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt)
-        sorted=true
+        processedPackageNames=$(cat $(Pipeline.Workspace)/pack/packagePublishOrder-${{ parameters.feedKind }}.txt)
       else
-        packages=*.tgz
-        sorted=false
+        # The sed expression removes from the first dash followed by a number, to the file extension.
+        # This assumes we don't have packages whose names have a dash followed by a number.
+        # See comment above for more details.
+        processedPackageNames=$(ls *.tgz | sed 's/-[0-9].*\.tgz//g')
```

Comment:
Please please please no... If we need to produce files that have package names only then lets do that and NOT write bash string manipulation logic.

Reply:
No objection here. I tried using `flub list` in eslint-config-fluid as a quick test and we'll need to update it to support something other than release groups and update `scripts/pack-packages.sh`. Also noticed that it fails if the provided outFile includes a path that doesn't exist (e.g. the feeds/ folder didn't exist in eslint-config-fluid), I think it should create the necessary folder structure itself.

Reply:
I tried coming back to this now that `flub list` supports package names as arguments but still can't leverage that functionality because the pipeline doesn't have access to a package name, we only have the `tagName` for the release, e.g. "eslint-config-fluid", so invoking `pack-packages.sh` with the right arguments isn't trivial.

I'm thinking if `flub list` might want to still take the `--releaseGroup` flag as an option, and if not provided default to the package in the current folder (withouth having to specify it). Then we could split the calls (or craft the list of arguments) to `flub list` inside pack-packages.sh based on [the existing check for the existence of `.releaseGroup` file](the existing check for the existence of `.releaseGroup` file).

Reply:
Not sure I follow... `flub list eslint-config-fluid` works. So does `flub list client` etc.

Reply:
Oh, I thought it was expecting an exact package name. And I see that it will find the correct package even if it lives in another folder, at least for some packages (it couldn't find telemetry-generator). Curious, does it do a full scan of the repo folder hierarchy until it finds a match? Do we have a mapping somewhere? Feels a bit like magic to me, which leaves me feeling a bit uneasy.

Also, seems like there's an implicit assumption/expectation somewhere that our `tagName` values are at least a substring match for the corresponding package (in the case of independent packages); if that's indeed the case, I'm not a huge fan,  but assuming that's true then I can see the way forward here.

Reply:
The build tools have a mapping of all the packages in the repo. See here for the logic that's used to find a package or release group based on the arg: https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-cli/src/args.ts

The convention of using the unscoped package name as the tag has been in place for a long long time. It's also used as the git tag name when we tag releases.

Reply:
I have an example change I've been testing with flub publish tarballs. I'll share it in the morning.

Reply:
#21018 demonstrates how I envision using flub publish tarballs. I tested things in dry-run mode but didn't try completely switching over.